fed/_private/compatible_utils.py

```diff
@@ -152,12 +153,28 @@ def reset(self):
 
 def _init_internal_kv():
     """An internal API that initialize the internal kv object."""
-    from ray._private.client_mode_hook import is_client_mode_enabled
-    if is_client_mode_enabled:
-        kv_actor = ray.remote(InternalKv).options(name="_INTERNAL_KV_ACTOR").remote()
-        response = kv_actor._ping.remote()
-        ray.get(response)
-    return ClientModeInternalKv() if is_client_mode_enabled else InternalKv()
-
-
-kv = _init_internal_kv()
+    global kv
+    if kv is None:
+        from ray._private.client_mode_hook import is_client_mode_enabled
+        if is_client_mode_enabled:
+            kv_actor = ray.remote(InternalKv) \
+                .options(name="_INTERNAL_KV_ACTOR").remote()
+            response = kv_actor._ping.remote()
+            ray.get(response)
+        kv = ClientModeInternalKv() if is_client_mode_enabled else InternalKv()
+
+
+def _clear_internal_kv():
+    global kv
+    if kv is not None:
+        kv.delete(constants.KEY_OF_CLUSTER_CONFIG)
```

Comment:
As in client mode, we put kv pairs in `internal_kv_actor`, should we clear them in internal_kv_actor as well?

Reply:
I think it's fine cuz, in client mode, kv is a `ClientModeInternalKv` which has inherited the `delete` interface.